app/src/main/java/org/vss/impl/postgres/PostgresBackendImpl.java

```diff
@@ -164,10 +167,14 @@ public ListKeyVersionsResponse listKeyVersions(ListKeyVersionsRequest request) {
       nextPageToken = keyVersions.get(keyVersions.size() - 1).getKey();
     }
 
-    return ListKeyVersionsResponse.newBuilder()
-        .setGlobalVersion(globalVersion)
+    ListKeyVersionsResponse.Builder responseBuilder = ListKeyVersionsResponse.newBuilder()
         .addAllKeyVersions(keyVersions)
-        .setNextPageToken(nextPageToken)
-        .build();
+        .setNextPageToken(nextPageToken);
+
+    if (Objects.nonNull(globalVersion)) {
+      responseBuilder.setGlobalVersion(globalVersion);
+    }
```

Comment:
I'm forgetting where we landed on this. How does the client interpret this value?

Reply:
Since clients need globalVersion to make subsequent successful writes, they need to update their client-side global_version for it.
Quoting: 
```
  // global_version is only returned in response for the first page of the ListKeyVersionsResponse
  // and is guaranteed to be read before reading any key-versions.
  //
  // In case of refreshing the complete key-version view on the client-side, correct usage for
  // the returned global_version is as following:
  //   1. Read global_version from the first page of paginated response and save it as local variable.
  //   2. Update all the key_versions on client-side from all the pages of paginated response.
  //   3. Update global_version on client_side from the local variable saved in step-1.
  // This ensures that on client-side, all current key_versions were stored at global_version or later.
  // This guarantee is helpful for ensuring the versioning correctness if using the global_version
  // in PutObject API and can help avoid the race conditions related to it.
```